### DISCUSSION TITLE
P vs NP Selección de estudiantes (Propuesta de Algoritmo)

### DISCUSSION THREAD

**Discussion by fsaraos27** [community/community#125608] [Programming Help]
### Select Topic Area

Show & Tell

### Body

**Problema de Selección de Estudiantes utilizando Teoría de Grafos y Algoritmos de Compatibilidad**

**Introducción**
La problemática de P vs NP ha sido un enigma en el campo de la computación durante décadas. En nuestra investigación, hemos abordado un problema NP práctico: la selección de estudiantes para alojamiento universitario con restricciones de incompatibilidad. Nuestro enfoque utiliza teoría de grafos y algoritmos de compatibilidad para encontrar soluciones eficientes.

**Descripción del Problema**
Contexto: Debemos organizar el alojamiento para 400 estudiantes, seleccionando solo 100 de ellos, sin emparejar a estudiantes incompatibles.

**Desafíos:**
La explosión combinatoria hace que encontrar una solución exacta sea impracticable.
Se necesitan algoritmos heurísticos para manejar eficientemente grandes volúmenes de datos.

**Nuestra Solución**
Datos Utilizados:
Identificador del Estudiante
Nombre
Edad
Género
Nacionalidad
Religión
Año de Estudio
Gustos Musicales
Actividades Extracurriculares
Hobbies

**Algoritmo Propuesto:**
**Construcción del Grafo:** Cada estudiante es un nodo y se crean aristas entre nodos si la compatibilidad supera un umbral definido.
**Cálculo de Compatibilidad:** Se evalúan varias características y se asignan puntos por coincidencia.
**Búsqueda de Cliques:** Usamos el algoritmo find_cliques de NetworkX para identificar subgrupos completamente conectados (cliques) que representan grupos de estudiantes altamente compatibles.
Resultados
**Eficiencia:** Nuestro algoritmo segmenta los datos en subgrupos manejables, mejorando significativamente el tiempo de procesamiento.
**Visualización:** Los resultados se presentan mediante grafos, validando visualmente la compatibilidad entre los estudiantes.
Impacto y Futuro
Nuestra investigación proporciona una base sólida para abordar problemas NP prácticos utilizando métodos heurísticos y teoría de grafos. Este enfoque puede ser adaptado para resolver otros problemas complejos en diversos campos.

**Conclusión:**
Invito a la comunidad a revisar mi trabajo, colaborar para mejorar los algoritmos y explorar nuevas aplicaciones de esta metodología.

**Enlace al Proyecto**
Para más detalles y acceder al código fuente, visita nuestro repositorio en GitHub: https://github.com/fsaraos27/p_vs_np

Mis datos de contacto:
fsaraos27@gmail.com
[Investigación](https://drive.google.com/file/d/1ZsuFRn-54kuZnbT2tQ1SaAW74thFroOG/view?usp=sharing)

[linkedin](https://www.linkedin.com/in/francisco-javier-saraos-garc%C3%ADa-38195a164/)
[Instagram](https://www.instagram.com/fsaraos/?hl=es)

**Comment by LiteBrite82** [comment]
Hi @fsaraos27 , thanks for participating!

Unfortunately, we don’t currently have moderators for languages other than English. Until that changes, we need to ask that everyone use English here in the GitHub Community when posting. We’ll be locking any posts in languages other than English for now, including this one.
